pkg/tcpip/stack/nic.go

```diff
@@ -502,7 +502,7 @@ func (n *NIC) DeliverNetworkPacket(linkEP LinkEndpoint, remote, _ tcpip.LinkAddr
 	// packet and forward it to the NIC.
 	//
 	// TODO: Should we be forwarding the packet even if promiscuous?
-	if n.stack.Forwarding() {
+	if n.stack.Forwarding(protocol) {
```

Comment:
I would rather not add another map lookup in this hot path. Can we use bools or an array instead? As for adding network protocol specific features, the IPv4Forwarding method on the stack seems pretty network protocol specific.

Reply:
Maybe it is better to use atomic operations here? 
https://gist.github.com/aleksej-paschenko/059ec6dda99177af6f110d61a39cd79b

Do you have some performance guidelines? Performance benchmarks for a network stack?

Reply:
We use atomic operations elsewhere and I think they could work well in this case.